crm_rma_prodlot_invoice/models/sale_order_line.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright 2015 Vauxoo
+#    Author: Yanina Aular, Osval Reyes
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import api, models
+
+
+class SaleOrderLine(models.Model):
+
+    _inherit = 'sale.order.line'
+
+    @api.multi
+    def invoice_line_create(self):
+        res = super(SaleOrderLine, self).\
+            invoice_line_create()
+        for order_line in self:
+            prodlot_obj = self.env['stock.production.lot']
+            procurements = order_line.procurement_ids
+            if procurements:
+                move_id = self.env['stock.move'].\
+                    search([('procurement_id', '=',
+                             order_line.procurement_ids[0].id)], limit=1)
+                if move_id and move_id.quant_ids:
+                    lot = move_id.quant_ids[0].lot_id
```

Comment:
For this functionality should not allow to use/set the same lot number for more than one product at time, we are about to use a module as dependency named [product_unique_serial](https://github.com/Vauxoo/stock-logistics-workflow/tree/8.0/product_unique_serial) to do that instead. This module it helps to do that what you mention. Thanks for noticing.